field_friend/automations/field.py

```diff
@@ -144,14 +144,24 @@ def _generate_rows(self) -> list[Row]:
 
     def _generate_outline(self) -> list[GeoPoint]:
         assert len(self.rows) > 0
-        outline_unbuffered: list[Point] = []
-        for p in self.rows[-1].points:
-            outline_unbuffered.append(p.cartesian())
-        outline_unbuffered.append(self.first_row_end.cartesian())
-        outline_unbuffered.append(self.first_row_start.cartesian())
-        outline_polygon = Polygon([p.tuple for p in outline_unbuffered])
-        bufferd_polygon = outline_polygon.buffer(
-            self.outline_buffer_width, join_style='mitre', mitre_limit=math.inf)
+        bufferd_polygon = None
+        if len(self.rows) > 1:
+            outline_unbuffered: list[Point] = []
+            for p in self.rows[-1].points:
+                outline_unbuffered.append(p.cartesian())
+            outline_unbuffered.append(self.first_row_end.cartesian())
+            outline_unbuffered.append(self.first_row_start.cartesian())
+            outline_polygon = Polygon([p.tuple for p in outline_unbuffered])
+            bufferd_polygon = outline_polygon.buffer(
+                self.outline_buffer_width, join_style='mitre', mitre_limit=math.inf)
+        else:
+            outline_unbuffered: list[Point] = []
+            for p in self.rows[0].points:
+                outline_unbuffered.append(p.cartesian())
+            outline_linestring = LineString([p.tuple for p in outline_unbuffered])
+            bufferd_polygon = outline_linestring.buffer(
+                self.outline_buffer_width, cap_style='square', join_style='mitre', mitre_limit=math.inf)
+    
```

Comment:
How about this to avoid duplicate code?
```suggestion
        outline_unbuffered: list[Point] = [
            self.first_row_end.cartesian(),
            self.first_row_start.cartesian()
        ]
        if len(self.rows) > 1:
            for p in self.rows[-1].points:
                outline_unbuffered.append(p.cartesian())
            outline_shape = Polygon([p.tuple for p in outline_unbuffered])
        else:
            outline_shape = LineString([p.tuple for p in outline_unbuffered])
        bufferd_polygon = outline_shape.buffer(
            self.outline_buffer_width, cap_style='square', join_style='mitre', mitre_limit=math.inf)
```